### ISSUE TITLE
Changing images during pyramid rendering causes QuPath to freeze

### ISSUE THREAD

**Issue by MarkZaidi** [qupath/qupath#1400]
**Describe the bug**
Opening a different image while a pyramid level is being rendered can cause QuPath to be unresponsive. A user can encounter this if they're quickly going through images searching for an example of some biological phenomenon (e.g. tumor infiltrating lymphocytes), which requires high resolution pyramid layers (level 0) to be rendered.

This is easier to reproduce on systems in which rendering is slower, such as projects with image files located on HDDs, images lacking pyramids, and low core count CPUs.

The images shown in the demo are 16-channel images from the COMET system, roughly 46.8 Gb in size, stored locally on an external HDD, and were converted to OME-TIFFs using the following script shown at the end of this ticket. There was a marginal decrease in instances of QuPath freezing when generating pyramids this way, rather than have bioformats handle it during import, so Pete's recommendation of pyramiding the images seems to have helped.

**To Reproduce**
Video recording with Visual VM process sampling: https://youtu.be/Pas61jyrWGQ?si=lYsq57oy5-pyANnU
1. Create a project with several large (>40Gb) multiplexed images
2. Quickly zoom in, then open another image
3. If QuPath doesn't become unresponsive, repeat the previous step with a new image, until it does.

**Expected behavior**
Any processes related to rendering of the closed image should be stopped, and QuPath should remain responsive

**Screenshots**

Hang on StopWaiting():
![StopWaiting screenshot](https://github.com/qupath/qupath/assets/52012166/05ac8527-76e4-4245-9cf5-9323681e6d2a)
Hang on FutureTask.get():
![FutureTask get screenshot](https://github.com/qupath/qupath/assets/52012166/c6ba65c6-268f-4202-83f9-782b7ed27035)

**Desktop (please complete the following information):**
 - OS: Windows 11 - 10.0
 - Version: 0.5.0-rc1 (able to reproduce on 0.4.3 too)
 - Build time: 2023-10-06, 13:28
 - Latest commit tag: '2241237'
 
**Additional context**
OME-TIFF conversion script:
```
// **** Save 32bit multichannel ICW image as OMETiff in QuPath 0.4.0_SNAPSHOT   ****

import javafx.application.Platform
import qupath.lib.scripting.QP
import qupath.lib.projects.Project
import qupath.lib.gui.scripting.QPEx
import qupath.lib.images.writers.ome.OMEPyramidWriter

//  ****** PARAMETERS *********

def outpath = "E:/pyramided images/"

// OME.Tiff parameters
def tilesize = 512
def outputDownsample = 1
def pyramidscaling = 2
def nThreads = 12
def compression = OMEPyramidWriter.CompressionType.LZW  //ZLIB //UNCOMPRESSED //LZW  (not working with 32bit: //J2K_LOSSY //J2K)

//  ***************************

//def imageData = QPEx.getCurrentViewer().getImageData()
def imageData = getCurrentImageData()
def server = imageData.getServer()
def name = getProjectEntry().getImageName()

def filename = name + ".ome.tif"
def pathOutput = outpath + filename

println('Writing OME-TIFF ' + filename)

new OMEPyramidWriter.Builder(server)
    .compression(compression)
    .parallelize(nThreads)
    .channelsInterleaved()     
    .tileSize(tilesize)
    .scaledDownsampling(outputDownsample, pyramidscaling)
    .build()
    .writePyramid(pathOutput)

println('Done:' + filename)
```
VisualVM forward calls:
[visualvm_forwardcalls.csv](https://github.com/qupath/qupath/files/13078670/visualvm_forwardcalls.csv)
.nps file of snapshots can't be uploaded to an issue, but I can provide it via email if needed.

**Comment by petebankhead** [comment]
Thanks, I have an idea what might be happening. I suspect it will only affect images being read using Bio-Formats.

**Comment by MarkZaidi** [comment]
I can't replicate it with any of the openslide .svs test images, so I think so too

**Comment by petebankhead** [comment]
Good. I haven't been able to replicate it using Bio-Formats either yet...

**Comment by petebankhead** [comment]
This was closed automatically when I merged the PR... but I *think* it's fixed. Feel free to reopen if you can replicate it in [v0.5.0-rc2](https://github.com/qupath/qupath/releases/tag/v0.5.0-rc2).

**Comment by MarkZaidi** [comment]
Seems to be fixed in v0.5.0-rc2 as I can no longer replicate it :thumbsup:

**Comment by petebankhead** [comment]
Great, thanks for confirming.
